### PR TITLE
Use channel-cp as seekPosition when FromDmlCPLoadDelete

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

issues: https://github.com/milvus-io/milvus/issues/21103

### CODE REVIEW

**Comment by yah01** [comment]
need to regenerate the mock_broker with `mockery`, as the interface changed

File: internal/datacoord/services.go [hunk 1]
```diff
@@ -347,6 +347,7 @@ func (s *Server) GetSegmentInfo(ctx context.Context, req *datapb.GetSegmentInfoR
 		return resp, nil
 	}
 	infos := make([]*datapb.SegmentInfo, 0, len(req.GetSegmentIDs()))
+	channelCPs := make(map[string]*internalpb.MsgPosition)
```
Comment (xiaofan-luan) [inline internal/datacoord/services.go:350]: add unit test

File: internal/querycoordv2/task/utils.go [hunk 2]
```diff
@@ -83,10 +83,16 @@ func packLoadSegmentRequest(
...
 	loadInfo *querypb.SegmentLoadInfo,
-	segment *datapb.SegmentInfo,
+	resp *datapb.GetSegmentInfoResponse,
 ) *querypb.LoadSegmentsRequest {
-	deltaPosition := segment.GetDmlPosition()
-	if deltaPosition == nil {
+	var deltaPosition *internalpb.MsgPosition
+	segment := resp.GetInfos()[0]
+
+	if resp.GetChannelCheckpoint() != nil && resp.ChannelCheckpoint[segment.InsertChannel] != nil {
```
Comment (xiaofan-luan) [inline internal/querycoordv2/task/utils.go:91]: if delta position is lagged(10 minutes or longer)
log here

**Pull request merged**